src/runtime_strings_extra.js

```diff
@@ -208,10 +208,14 @@ function lengthBytesUTF32(str) {
 // Allocate heap space for a JS string, and write it there.
 // It is the responsibility of the caller to free() that memory.
 function allocateUTF8(str) {
+#if '_malloc' in IMPLEMENTED_FUNCTIONS
   var size = lengthBytesUTF8(str) + 1;
   var ret = _malloc(size);
```

Comment:
I'm surprised not to see the pattern used more in the library code?  Is that idea to wrap all malloc calls in `#if '_malloc' in IMPLEMENTED_FUNCTIONS`

I wonder if this could be rewritten as:

```
var size = lengthBytesUTF8(str) + 1;
var ret = {{{ makeMalloc("allocateUTF8") }}} (size);
if (ret) stringToUTF8Array(str, HEAP8, ret, size);
return ret;
```

Thus avoiding the ifdef here?
var size = lengthBytesUTF8(str) + 1;
```

Reply:
Thanks, good idea to use a macro here, done.

This doesn't happen more because the deps_info mechanism is usually enough. But in 2 places we can't use deps (in the FS we don't want any FS usage to force malloc, and allocUTF8 isn't tracked by deps - so in both cases we don't have the fine-grained tracking we need for deps_info).